Src/BootCamp.Chapter/Factory.cs

```diff
@@ -0,0 +1,34 @@
+﻿using BootCamp.Chapter.Computer;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    public abstract class Factory
+    {
+        protected Factory() { }
+
+        public DesktopComputer Assemble()
+        {
+            DesktopComputer computer = new DesktopComputer();
```

Comment:
At first, I wanted to make the Assemble method abstract thus both ms and mac factories would have had different implementations.

But seeing that I had to go through the same steps when the assembly happens, I've made the steps abstract methods instead.

Again, this could probably be done in a much more elegant way and components should be handled to the Assemble method so it's not statically coded inside the abstract methods.

Reply:
That was the point of the exercise. The exercise explicitly states "Both factories have the same flow of assembling parts, just individual steps are different".
So yes, this is the correct approach.